File: funcoes.cpp
#include <iostream>
#include <vector>
using namespace std;

string verificar_estoque_critico(int quantidade) {
    if (quantidade < 5) {
        return " [REPOSICAO NECESSARIA]";
    }
    return "";
}

void exibir_cabecalho() {
    cout << "   SORVETERIA DO DENER | CONTROLE DE ESTOQUE   \n";
}

int main() {
    vector<string> nomes;
    vector<int> quantidades;

    for (int i = 0; i < 3; i++) {
        string nome;
        int qtd;

        cout << "Digite o nome do produto " << i+1 << ": ";
        cin >> nome;

        do {
            cout << "Digite a quantidade de " << nome << ": ";
            cin >> qtd;

            if (qtd < 0) {
                cout << "Erro: o estoque nao pode ser negativo. Tente novamente.\n";
            }

        } while (qtd < 0);

        nomes.push_back(nome);
        quantidades.push_back(qtd);
    }

    exibir_cabecalho();

    cout << "\nRELATORIO FINAL\n";

    int i = 0;
    while (i < nomes.size()) {
        string aviso = verificar_estoque_critico(quantidades[i]);

        cout << "Produto: " << nomes[i] << aviso
             << " | Estoque: " << quantidades[i] << " unidades\n";

        i++;
    }

    return 0;
}

File: funcoes.java
import java.util.ArrayList;
import java.util.Scanner;

public class Main {

    public static String verificar_estoque_critico(int quantidade) {
        if (quantidade < 5) {
            return " [REPOSICAO NECESSARIA]";
        }
        return "";
    }

    public static void exibir_cabecalho() {
        System.out.println("   SORVETERIA DO DENER 🍦 | CONTROLE DE ESTOQUE     ");
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        ArrayList<String> nomes = new ArrayList<>();
        ArrayList<Integer> quantidades = new ArrayList<>();

        for (int i = 0; i < 3; i++) {
            System.out.print("Digite o nome do produto " + (i + 1) + ": ");
            String nome = sc.next();

            int qtd;
            do {
                System.out.print("Digite a quantidade de " + nome + ": ");
                qtd = sc.nextInt();

                if (qtd < 0) {
                    System.out.println("Erro: o estoque nao pode ser negativo. Tente novamente.");
                }

            } while (qtd < 0);

            nomes.add(nome);
            quantidades.add(qtd);
        }

        exibir_cabecalho();

        System.out.println("\nRELATORIO FINAL");

        int i = 0;
        while (i < nomes.size()) {
            String aviso = verificar_estoque_critico(quantidades.get(i));

            System.out.println("Produto: " + nomes.get(i) + aviso +
                    " | Estoque: " + quantidades.get(i) + " unidades");

            i++;
        }

        sc.close();
    }
}